website/docs/howto.md

```diff
@@ -0,0 +1,278 @@
+that would result in bypassing constraints for the namespace. The namespace-label webhook
+is necessary to prevent a privilege escalation where the permission to add a label to a
+namespace is equivalent to the ability to bypass all constraints for that namespace.
+You can read more about the ability to exempt namespaces by label [above](#exempting-namespaces-from-the-gatekeeper-admission-webhook).
+
+Because Kubernetes adds features with each version, if you want to know how the webhook can be configured it
+is best to look at the official documentation linked at the top of this section. However, two particularly important
+configuration options deserve special mention: [timeouts](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#timeouts) and
+[failure policy](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#failure-policy).
+
+Timeouts allow you to configure how long the API server will wait for a response from the admission webhook before it
+considers the request to have failed. Note that setting the timeout longer than the overall request timeout
+means that the main request will time out before the webhook's failure policy is invoked.
+
+Failure policy controls what happens when a webhook fails for whatever reason. Common
+failure scenarios include timeouts, a 5xx error from the server or the webhook being unavailable.
+You have the option to ignore errors, allowing the request through, or failing, rejecting the request.
+This results in a direct tradeoff between availability and enforcement.
+
+Currently Gatekeeper is defaulting to using `Ignore` for the constraint requests. This is because
+the webhook server currently only has one instance, which risks downtime during actions like upgrades.
+As the theoretical availability improves we will likely change the default to `Fail`.
+
+The namespace label webhook defaults to `Fail`, this is to help ensure that policies preventing
+labels that bypass the webhook from being applied are enforced. Because this webhook only gets
+called for namespace modification requests, the impact of downtime is mitigated, making the
+theoretical maximum availability less of an issue.
+
+Because the manifest is available for customization, the webhook configuration can
+be tuned to meet your specific needs if they differ from the defaults.
+
+### Emergency Recovery
+
+If a situation arises where Gatekeeper is preventing the cluster from operating correctly,
+the webhook can be disabled. This will remove all Gatekeeper admission checks. Assuming
+the default webhook name has been used this can be achieved by running:
+
+`kubectl delete validatingwebhookconfigurations.admissionregistration.k8s.io gatekeeper-validating-webhook-configuration`
+
+Redeploying the webhook configuration will re-enable Gatekeeper.
+
+### Running on private GKE Cluster nodes
```

Comment:
We should have a separate page for cloud-specific fixes.

Reply:
splitted into cloud and vendor specific fixes page